File: python/difference-of-squares/difference_of_squares.py
from functools import reduce
def square_of_sum(count):
    return sum(range(1, count + 1)) ** 2

def sum_of_squares(count):
    return reduce(lambda s, n: s + n ** 2, range(1, count + 1), 0)


def difference(count):
    return square_of_sum(count) - sum_of_squares(count)

File: dart/difference-of-squares/lib/difference_of_squares.dart
class DifferenceOfSquares {
  int squareOfSum(int n) {
    int s = (n * (n + 1)) ~/ 2;
    return s*s;
  }

  int sumOfSquares(int n) {
    return (new List<int>.generate(n, (i) => i + 1)).reduce((memo, element) => memo + element*element);
  }

  int differenceOfSquares(int n) {
    return squareOfSum(n) - sumOfSquares(n);
  }
}
